tests/components/sensibo/test_coordinator.py

```diff
@@ -0,0 +1,66 @@
+"""The test for the sensibo coordinator."""
+from __future__ import annotations
+
+from unittest.mock import patch
+
+from pysensibo.exceptions import AuthenticationError, SensiboError
+from pysensibo.model import SensiboData
+import pytest
+
+from homeassistant.components.sensibo.const import DOMAIN
+from homeassistant.components.sensibo.coordinator import SensiboDataUpdateCoordinator
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed
+from homeassistant.helpers.update_coordinator import UpdateFailed
+
+from . import init_integration
+from .response import DATA_FROM_API
+
+
+async def test_coordinator(hass: HomeAssistant) -> None:
+    """Test the Sensibo coordinator."""
+    entry = await init_integration(hass, entry_id="hallcd")
+    with patch(
+        "homeassistant.components.sensibo.coordinator.SensiboClient.async_get_devices_data",
+        return_value=DATA_FROM_API,
+    ):
+        await hass.config_entries.async_setup(entry.entry_id)
+    await hass.async_block_till_done()
+
+    coordinator: SensiboDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+    assert coordinator.data.parsed["ABC999111"].state == "heat"
+
+
+async def test_coordinator_errors(hass: HomeAssistant) -> None:
+    """Test the Sensibo coordinator errors."""
+    entry = await init_integration(hass, entry_id="hallcd2")
+    with patch(
+        "homeassistant.components.sensibo.coordinator.SensiboClient.async_get_devices_data",
+        return_value=DATA_FROM_API,
+    ):
```

Comment:
This repeated many times. Many make this a pytest fixture?

Reply:
I will do that. I was going back and forth with these ones.
I thought pytest "reset" between the tests which they appear to do here in the tests but not locally in my dev container.

Reply:
It depends how the tests run, if you use the coverage task in the dev container they run in parallel similar to how they run in Github actions, if you run pytest manually you run them one after another. In any case to make sure you don't change the fixture during test you can use `moneypatch` so your changes are undone when the test finished.

https://docs.pytest.org/en/6.2.x/monkeypatch.html
Example: https://github.com/home-assistant/core/blob/1e18307a6678b6f77bf53567c4aa22d005cf190b/tests/components/switcher_kis/test_sensor.py#L88

Reply:
Interesting. Good to know but in this case I'm calling services and thereby changing the data so when tests run sequential it affect the test following it.
But now I know how it operates in this sense so I can deal with it.

Reply:
I am not sure if it is the right approach but what I have seen other integrations doing is making a copy of the initial data and working on this copy so that you always start from the same data.

Reply:
Tried that too from a common function but apparently it seems as the tests "cache" the function in question and thereby also the data gets shared.
I will see what I do in the end but I guess I just end up with dealing with the fact it's sharing the same data (work it in as intended).

Reply:
You might need to change the scope on a fixture
https://docs.pytest.org/en/7.1.x/how-to/fixtures.html#scope-sharing-fixtures-across-classes-modules-packages-or-session

Reply:
Was looking at that too but wasn't using it as a fixture but I will give it a try to reduce the scope of it.